### PR TITLE
sync ui geometry.dart into web_ui geometry.dart; add null safety annotations

### PR DESCRIPTION
**Pull request by yjbanov**

### CODE REVIEW

File: lib/ui/geometry.dart [hunk 1]
```diff
@@ -13,14 +13,12 @@ abstract class OffsetBase {
   ///
   /// The first argument sets the horizontal component, and the second the
   /// vertical component.
-  const OffsetBase(double dx, double dy)
-      : _dx = dx ?? 0.0,
-        _dy = dy ?? 0.0,
```
Comment (cbracken) [inline lib/ui/geometry.dart:18]: Wow -- this was for the STRAIGHT TO RELEASE MODE crowd I guess :)

File: lib/ui/geometry.dart [hunk 2]
```diff
@@ -78,24 +76,24 @@ abstract class OffsetBase {
...
         && other._dy == _dy;
   }
 
   @override
-  int get hashCode => hashValues(_dx, _dy);
+  int/*!*/ get hashCode => hashValues(_dx, _dy);
 
   @override
-  String toString() => 'OffsetBase(${_dx?.toStringAsFixed(1)}, ${_dy?.toStringAsFixed(1)})';
+  String/*!*/ toString() => 'OffsetBase(${_dx?.toStringAsFixed(1)}, ${_dy?.toStringAsFixed(1)})';
```
Comment (cbracken) [inline lib/ui/geometry.dart:96]: Shouldn't we kill the `?` here?

File: lib/ui/geometry.dart [hunk 3]
```diff
@@ -122,37 +120,37 @@ abstract class OffsetBase {
 class Offset extends OffsetBase {
   /// Creates an offset. The first argument sets [dx], the horizontal component,
   /// and the second sets [dy], the vertical component.
-  const Offset(double dx, double dy) : super(dx, dy);
+  const Offset(double/*!*/ dx, double/*!*/ dy) : super(dx, dy);
```
Comment (cbracken) [inline lib/ui/geometry.dart:123]: Optionally, add non-null asserts for dx/dy here for (hopefully very limited-time) future proofing.

File: lib/ui/geometry.dart [hunk 4]
```diff
@@ -329,43 +327,43 @@ class Offset extends OffsetBase {
...
         && other.dy == dy;
   }
 
   @override
-  int get hashCode => hashValues(dx, dy);
+  int/*!*/ get hashCode => hashValues(dx, dy);
 
   @override
-  String toString() => 'Offset(${dx?.toStringAsFixed(1)}, ${dy?.toStringAsFixed(1)})';
+  String/*!*/ toString() => 'Offset(${dx?.toStringAsFixed(1)}, ${dy?.toStringAsFixed(1)})';
```
Comment (cbracken) [inline lib/ui/geometry.dart:340]: Kill the `?`s here.

File: lib/ui/geometry.dart [hunk 5]
```diff
@@ -595,17 +593,17 @@ class Size extends OffsetBase {
...
         && other._dy == _dy;
   }
 
   @override
-  int get hashCode => hashValues(_dx, _dy);
+  int/*!*/ get hashCode => hashValues(_dx, _dy);
 
   @override
-  String toString() => 'Size(${width?.toStringAsFixed(1)}, ${height?.toStringAsFixed(1)})';
+  String/*!*/ toString() => 'Size(${width?.toStringAsFixed(1)}, ${height?.toStringAsFixed(1)})';
```
Comment (cbracken) [inline lib/ui/geometry.dart:606]: Kill the `?`s here too.

File: lib/ui/geometry.dart [hunk 6]
```diff
@@ -329,43 +327,43 @@ class Offset extends OffsetBase {
...
+  String/*!*/ toString() => 'Offset(${dx?.toStringAsFixed(1)}, ${dy?.toStringAsFixed(1)})';
 }
 
 /// Holds a 2D floating-point size.
 ///
 /// You can think of this as an [Offset] from the origin.
 class Size extends OffsetBase {
   /// Creates a [Size] with the given [width] and [height].
-  const Size(double width, double height) : super(width, height);
+  const Size(double/*!*/ width, double/*!*/ height) : super(width, height);
```
Comment (cbracken) [inline lib/ui/geometry.dart:348]: Again optionally add the asserts.

File: lib/ui/geometry.dart (hunk 2 above)
Comment (yjbanov) [reply lib/ui/geometry.dart:96]: Done.

File: lib/ui/geometry.dart (hunk 3 above)
Comment (yjbanov) [reply lib/ui/geometry.dart:123]: The `super` already does that (as of this PR).

File: lib/ui/geometry.dart (hunk 4 above)
Comment (yjbanov) [reply lib/ui/geometry.dart:340]: The tool actually did it automatically! But removed anyway.

File: lib/ui/geometry.dart (hunk 6 above)
Comment (yjbanov) [reply lib/ui/geometry.dart:348]: Same here: see the implementation of `OffsetBase`.

File: lib/ui/geometry.dart (hunk 5 above)
Comment (yjbanov) [reply lib/ui/geometry.dart:606]: Done.

**Comment by yjbanov** [comment]
The pending bots are infra flakiness. Merging.

**Pull request merged**